From e5423da4d7ca010fa3c33736445752ff9e549585 Mon Sep 17 00:00:00 2001
From: Dominique Leuenberger <dimstar@opensuse.org>
Date: Tue, 7 Aug 2012 18:10:24 +0200
Subject: [PATCH] cdda: Support libcdio 0.84 with changed API

following CDIO changeset
http://git.savannah.gnu.org/gitweb/?p=libcdio.git;a=commitdiff;h=abe5b8429573c9ac796c274c7abe7f76ef52dc76

https://bugzilla.gnome.org/show_bug.cgi?id=678257

[Alexandre Rostovtsev <tetromino@gentoo.org> : cherry-pick for 1.12]
---
 daemon/gvfsbackendcdda.c | 32 +++++++++++++++++++++++++-------
 1 file changed, 25 insertions(+), 7 deletions(-)

diff --git a/daemon/gvfsbackendcdda.c b/daemon/gvfsbackendcdda.c
index 6f009c7..1ad4251 100644
--- a/daemon/gvfsbackendcdda.c
+++ b/daemon/gvfsbackendcdda.c
@@ -169,11 +169,22 @@ fetch_metadata (GVfsBackendCdda *cdda_backend)
   if (!cdio)
     return;
 
-  cdtext = cdio_get_cdtext(cdio, 0);
+#if LIBCDIO_VERSION_NUM >= 84
+  cdtext = cdio_get_cdtext (cdio);
+#else
+  cdtext = cdio_get_cdtext (cdio, 0);
+#endif /* LIBCDIO_VERSION_NUM >= 84 */
+
   if (cdtext) {
-    cdda_backend->album_title = g_strdup (cdtext_get (CDTEXT_TITLE, cdtext));
-    cdda_backend->album_artist = g_strdup (cdtext_get (CDTEXT_PERFORMER, cdtext));
-    cdda_backend->genre = g_strdup (cdtext_get (CDTEXT_GENRE, cdtext));
+#if LIBCDIO_VERSION_NUM >= 84
+    cdda_backend->album_title = g_strdup (cdtext_get_const (cdtext, CDTEXT_FIELD_TITLE, 0));
+    cdda_backend->album_artist = g_strdup (cdtext_get_const (cdtext, CDTEXT_FIELD_PERFORMER, 0));
+    cdda_backend->genre = g_strdup (cdtext_get_const (cdtext, CDTEXT_FIELD_GENRE, 0));
+#else
+    cdda_backend->album_title = g_strdup (cdtext_get_const (CDTEXT_TITLE, cdtext));
+    cdda_backend->album_artist = g_strdup (cdtext_get_const (CDTEXT_PERFORMER, cdtext));
+    cdda_backend->genre = g_strdup (cdtext_get_const (CDTEXT_GENRE, cdtext));
+#endif /* LIBCDIO_VERSION_NUM >= 84 */
   }
 
   cdtrack = cdio_get_first_track_num(cdio);
@@ -182,10 +193,17 @@ fetch_metadata (GVfsBackendCdda *cdda_backend)
   for ( ; cdtrack < last_cdtrack; cdtrack++ ) {
     GVfsBackendCddaTrack *track;
     track = g_new0 (GVfsBackendCddaTrack, 1);
-    cdtext = cdio_get_cdtext(cdio, cdtrack);
+#if LIBCDIO_VERSION_NUM < 84
+    cdtext = cdio_get_cdtext (cdio, cdtrack);
+#endif /* LIBCDIO_VERSION_NUM < 84 */
     if (cdtext) {
-      track->title = g_strdup (cdtext_get (CDTEXT_TITLE, cdtext));
-      track->artist = g_strdup (cdtext_get (CDTEXT_PERFORMER, cdtext));
+#if LIBCDIO_VERSION_NUM >= 84
+      track->title = g_strdup (cdtext_get_const (cdtext, CDTEXT_FIELD_TITLE, cdtrack));
+      track->artist = g_strdup (cdtext_get_const (cdtext, CDTEXT_FIELD_PERFORMER, cdtrack));
+#else
+      track->title = g_strdup (cdtext_get_const (CDTEXT_TITLE, cdtext));
+      track->artist = g_strdup (cdtext_get_const (CDTEXT_PERFORMER, cdtext);
+#endif /* LIBCDIO_VERSION_NUM >= 84 */
     }
     track->duration = cdio_get_track_sec_count (cdio, cdtrack) / CDIO_CD_FRAMES_PER_SEC;
 
-- 
1.9.1

